rope.c

```diff
@@ -211,6 +290,72 @@ static size_t count_bytes_in_utf8(const uint8_t *str, size_t num_chars) {
   return p - str;
 }
 
+/// Skip to the given UTF-8 offset in the string.
+/// \param str The string to search.
+/// \param chars The number of characters to skip.
+/// \param bytelen The length of the string in bytes.
+/// \return A pointer to the new offset in the string.
+static inline uint8_t *skip_utf8(uint8_t *str, size_t chars, size_t bytelen) {
```

Comment:
I think there's already code to do this stuff in the project. Seems like duplication adding this here.

Reply:
Yep use `count_bytes_in_utf8` instead.